presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -165,6 +171,8 @@ public Page getOutput()
                 if (!advanceProbePosition()) {
                     break;
                 }
+                statisticsCounter.recordProbe(joinSourcePositions);
```

Comment:
This will be inlined by JIT. If we consider code of recordProbe it contains 3 conditional jumps and reference to arrays of size 8 * 2 * 8=2 * 64=2 cache lines, fits L1. According to Jeff Dean numbers, max cost for 3 branch misprediction and 5*3+0.5=15.5ns. I think 1 mispredict actually, because code generated will look like (conditional mov that is not a branch, one actual branch - all C++ compilers do that for years to eliminate branches, I think Java JIT is good enough too):
```
mov numSourcePosition, %eax
cmp %eax, 10
jump_gt GT_10
# here %eax <= 10
cmp %eax, 4
movle %eax, %edx
movgt 5, %edx
jump END_IF
GT_10:
cmp %eax, 100
movle 6, %edx
movgt 7, %edx
END_IF:
...
```
So actual cost is 5.5ns which is less than L2 cache reference.

Compared to random access of a hashmap of build size (100ns best!) and then byte copying this is tiny. AL ops cost nothing. Can do JMH, but it's true 0 overhead compared to probe copy, hashmap lookup (don't forget equality check), build copy and etc. Having percentile digest is a question (and I didn't chose it - may be expensive).